plugins/ccgx/fu-ccgx-dmc-device.c

```diff
@@ -0,0 +1,703 @@
+/*
+ * Copyright (C) 2020 Cypress Semiconductor Corporation.
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-chunk.h"
+
+#include "fu-ccgx-common.h"
+#include "fu-ccgx-dmc-common.h"
+#include "fu-ccgx-dmc-device.h"
+#include "fu-ccgx-dmc-firmware.h"
+
+#define DMC_FW_WRITE_STATUS_RETRY_DELAY_MS 30 /* 30 msec */
+
+struct _FuCcgxDmcDevice {
+	FuUsbDevice		parent_instance;
+	FWImageType		fw_image_type;
+        DmcDockIdentity		dock_id;
+        guint8			ep_intr_in;
+        guint8			ep_bulk_out;
+	guint32			fw_data_written;
+	guint32			fw_data_size;
+	DmcUpdateModel 		update_model;
+};
+
+G_DEFINE_TYPE (FuCcgxDmcDevice, fu_ccgx_dmc_device, FU_TYPE_USB_DEVICE)
+
+static gboolean
+fu_ccgx_dmc_device_get_dock_id (FuCcgxDmcDevice *self,
+			        DmcDockIdentity* dock_id,
+			        GError **error)
+{
+	if (dock_id == NULL) {
+		g_set_error(error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "invalid  pointer at %s",__func__);
+			return FALSE;
```

Comment:
Is this better to use `g_return_val_if_fail` ?

Reply:
Fixed on today commit (ccgx: fix code alignment, modify code for review)